tests/src/Core/Config/Cache/ConfigCacheTest.php

```diff
@@ -275,4 +275,36 @@ public function testKeyDiffWithoutResult($data)
 
 		$this->assertEmpty($configCache->keyDiff($diffConfig));
 	}
+
+	/**
+	 * Test the default hiding of passwords inside the cache
+	 */
+	public function testPasswordHide()
+	{
+		$configCache = new ConfigCache([
+			'database' => [
+				'password' => 'supersecure',
+				'username' => 'notsecured',
+			],
+		]);
+
+		$this->assertNotEquals('supersecure', print_r($configCache->get('database', 'password'), true));
```

Comment:
How do you retrieve the config value then? Isn't `HiddenString->__toString` kicking in?

Reply:
`print_r` uses a debugging way to print the value, this causes hiddenstring to wipe the the value. The normal usage isn't blocked.

=> I checked both tests
=> Added a counter-check to prove it :-)

Reply:
Awesome sauce!